libs/feature/search/src/lib/state/effects.ts

```diff
@@ -130,17 +130,19 @@ export class SearchEffects {
             }
             return this.filterGeometry$.pipe(
               tap((geom) => {
-                try {
-                  const trace = validGeoJson(geom, true) as string[]
-                  if (trace?.length > 0) {
-                    throw new Error(trace.join('\n'))
+                if (geom) {
+                  try {
+                    const trace = validGeoJson(geom, true) as string[]
+                    if (trace?.length > 0) {
+                      throw new Error(trace.join('\n'))
+                    }
+                  } catch (error) {
+                    console.warn(
+                      'Error while parsing the geometry filter\n',
+                      error
+                    )
+                    throw new Error()
                   }
-                } catch (error) {
-                  console.warn(
-                    'Error while parsing the geometry filter\n',
-                    error
-                  )
-                  throw new Error()
                 }
               }),
               map((geom) => [state, favorites, geom]),
```

Comment:
Was just wondering if it was better to return `null` here (like in the `catchError` below) if the `geom` is `undefined`...
```suggestion
              map((geom) => [state, favorites, geom ?? null]),
```

Reply:
yes that sounds good, it would mean we would reliably get `null` for a geometry in all those cases

Reply:
Actually, I noticed that we are converting nullish geometries to `undefined` [further down](https://github.com/geonetwork/geonetwork-ui/pull/798/files#diff-3d941fbb88225a34e9370b338aaf79c8c0fb0e4f4cdb5f060932f5d3e995ee36L173), so I'm hesitating a little where to use `null` and `undefined`. Returning `undefined` in line [173](https://github.com/geonetwork/geonetwork-ui/pull/798/files#diff-3d941fbb88225a34e9370b338aaf79c8c0fb0e4f4cdb5f060932f5d3e995ee36L173) seems reasonable, as it's a new object and we do the same for `filterIds`. So I kept `null` in the modified `switchMap` and `undefined` in the following. What do you think?

Reply:
So as discussed, I consistently put `undefined` in the end, which should make the code more readable.